analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -897,6 +897,27 @@ public void CatchWhen_Finally()
             validator.TagStates("End").Should().HaveCount(1).And.ContainSingle(x => HasNoException(x));
         }
 
+        [TestMethod]
+        public void Finally_NestedLambda_ShouldNotFail()
+        {
+            const string code = @"
+Tag(""Unreachable - outer CFG is not analyzed"");
+try
+{
+    Action a = () =>
+    {
+        var tag = ""Before"";
+        Tag(""CanThrow"");
+        tag = ""After"";
+    };
+}
+finally
+{
+    var nonEmptyFinallyBlock = true;
+}";
```

Comment:
I would add more `Tag(""Unreachable - outer CFG is not analyzed"");`
* before line 907
* after line 912
* after line 917
* instead of line 916

Reply:
I put it instead of 916, as that one was a bit artificial. The rest is not needed because of the 1st one.